src/vm/eventpipe.cpp

```diff
@@ -341,6 +341,8 @@ void EventPipe::Disable(EventPipeSessionID id)
     }
     CONTRACTL_END;
 
+    SetupThread();
```

Comment:
Sorry for the potential dummy question, but do we understand that this function will not have negative side effects when invoking it multiple times? Is there a counter part function we need to call to "tear down" the thread?

Reply:
The fix/change seems trivial though.

Reply:
I was able to run the `gcperfsim` with my hacked version of `dotnet trace` that enable and disable the section 100 times without any observable side effects. I think we are good.